File: Backend/Backend.API/Data/DTOs/AddOrderRequest.cs
using System.Text.Json.Serialization;
using Backend.API.Data.Enums;
using Backend.API.Data.Models;

namespace Backend.API.Data.DTOs
{
    public class AddOrderRequest
    {
        public int PizzaId { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public PizzaSize PizzaSize { get; set; }
        public ICollection<int> ToppingIds { get; set; }

        public Order ToEntity()
        {
            var order = new Order
            {
                PizzaId = PizzaId,
                PizzaSize = PizzaSize,
            };

            List<OrderTopping> orderToppings = new List<OrderTopping>();
            foreach (var toppingId in ToppingIds)
            {
                orderToppings.Add(new OrderTopping { ToppingId = toppingId });
            }
            order.OrderToppings = orderToppings;

            return order;
        }
    }
}

File: frontend/src/data/dtos/AddOrderRequest.ts
import { PizzaSize } from "./OrderResponse";

export class AddOrderRequest {
  pizzaId: number;
  pizzaSize: PizzaSize;
  toppingIds: number[];

  constructor(pizzaId: number, pizzaSize: PizzaSize, toppingIds: number[]) {
    this.pizzaId = pizzaId;
    this.pizzaSize = pizzaSize;
    this.toppingIds = toppingIds;
  }
}
